### PR TITLE
fix: add repoutils.FindCreateRepoCommand to flexibly use either creat…

### PR DESCRIPTION
**Pull request by SeanDougherty**
…erepo or createrepo_c

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Improves our retrieval of the createrepo command used by the rpmrepomanager. 

Previously, running image customizer on an Ubuntu devbox resulted in a `executable file not found in $PATH error` even with createrepo-c (Debian's createrepo package) installed.

Screenshot below shows:
* repro of the error
* building the tool with this fix
* a successful run with the tool now fixed
image

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- local

### CODE REVIEW

File: toolkit/tools/internal/packagerepo/repomanager/rpmrepomanager/rpmrepomanager.go [hunk 1]
```diff
@@ -36,8 +37,14 @@ func CreateRepo(repoDir string) (err error) {
 		return
 	}
 
+	createRepoCmd, err := repoutils.FindCreateRepoCommand()
+	if err != nil {
+		logger.Log.Warn(err)
```
Comment (dmcilvaney) [inline toolkit/tools/internal/packagerepo/repomanager/rpmrepomanager/rpmrepomanager.go:42]:
```suggestion
		return fmt.Errorf("unable to create repo:\n%w", err)
```

File: toolkit/tools/internal/packagerepo/repomanager/rpmrepomanager/rpmrepomanager.go [hunk 2]
```diff
@@ -48,8 +55,15 @@ func CreateRepo(repoDir string) (err error) {
 // CreateOrUpdateRepo will create an RPM repository at repoDir or update
 // it if the metadata files already exist.
 func CreateOrUpdateRepo(repoDir string) (err error) {
+	// Check if createrepo command is available
+	createRepoCmd, err := repoutils.FindCreateRepoCommand()
+	if err != nil {
+		logger.Log.Warn(err)
```
Comment (dmcilvaney) [inline toolkit/tools/internal/packagerepo/repomanager/rpmrepomanager/rpmrepomanager.go:61]:
```suggestion
		return fmt.Errorf("unable to create repo:\n%w", err)
```

File: toolkit/tools/internal/packagerepo/repomanager/rpmrepomanager/rpmrepomanager.go [hunk 3]
```diff
@@ -36,8 +37,14 @@ func CreateRepo(repoDir string) (err error) {
 		return
 	}
 
+	createRepoCmd, err := repoutils.FindCreateRepoCommand()
+	if err != nil {
+		return fmt.Errorf("unable to create repo:\n%w", err)
+		return
```
Comment (cwize1) [inline toolkit/tools/internal/packagerepo/repomanager/rpmrepomanager/rpmrepomanager.go:43]: Double return statement. Same problem below.

File: toolkit/tools/internal/packagerepo/repoutils/repoutils.go [hunk 4]
```diff
@@ -157,3 +158,24 @@ func verifyClonedRepoContents(clonedRepoContents, expectedPackages []*repocloner
...
+	for _, cmd := range creatrepo_cmds {
+		_, err = exec.LookPath(cmd)
+		if err == nil {
+			selectedCmd = cmd
+			break
+		}
+	}
+
+	if selectedCmd == "" {
+		return "", fmt.Errorf("failed to find a working createrepo command. Attempted commands: %v", creatrepo_cmds)
```
Comment (cwize1) [inline toolkit/tools/internal/packagerepo/repoutils/repoutils.go:177]: nit: `failed to find a working createrepo command:\nattempted commands: %v`

**Comment by cwize1** [comment]
Should be targeting the `3.0-dev` branch.

**Review by cwize1 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**